src/reference.rs

```diff
@@ -19,6 +19,20 @@ impl<T> ReferenceOr<T> {
     pub fn boxed_item(item: T) -> ReferenceOr<Box<T>> {
         ReferenceOr::Item(Box::new(item))
     }
+
+    pub fn into_item(self) -> Option<T> {
+        match self {
+            ReferenceOr::Reference { reference: _ } => None,
```

Comment:
nit: I think this could be 
```suggestion
            ReferenceOr::Reference { .. } => None,
```